tool/plugin/pubspec.yaml

```diff
@@ -0,0 +1,14 @@
+name: plugin
```

Comment:
This PR is creating a new `tool/plugin/` sub-dir and module. There's already a pubspec.yaml at the top level, and some dart code in `tool/`. It seems like we should use that existing dart project, put the `plugin` cli code in tool, and - more medium term - plan to migrate the current ad-hoc script in tool/ to be driven through `plugin` (currently, they re-generate our support for displaying flutter icons and colors in the IDE).

Thoughts?

Reply:
When I started working on this I was getting frustrated by the lack of Dart editing support. Rather than trying to solve the project configuration problem (which I did spend some time on) I decided to side-step it by generating a properly-configured module.

Merging existing tool functionality into the plugin tool makes sense. That looks straight-forward.

I think its cleaner to keep Java code in a Java project and use a Dart module to handle Dart code. We could remove pubspec.yaml and friends from the top level once we've merged everything. But, I'm happy to rearrange things the way you suggest if the project is configured to support Dart editing -- sorting out the configuration issues was a distraction I didn't want.

Reply:
It is perhaps messier (to combine both Dart and Java). Happy either way I suppose.

We may want to consider a re-org of the repo a bit. I think it might be a bit cleaner to have 3 sibling modules (rather than one at the root and two - AS and Dart - as children). That would be a big re-org, so may not be justified.